### DISCUSSION TITLE
What are the steps to use this with a Godot 4.3 build?

### DISCUSSION THREAD

**Discussion by 31** [quaadgras/graphics.gd#41] [Q&A] [answered]
I see Godot 4.2.2 mentioned throughout the project and docs, but I'm not sure what makes it specific to that version. What's the process to update to a new version? (Is there a doc I missed?)

For now, I have a build of Godot 4.3 that I'd like to use. But even using official Godot builds, in the future I might want to downgrade if I want to stay on an old Godot minor version for stability.

**Comment by Splizard** [comment]
You will likely be able to include a Go extension built with `gd` in a Godot 4.3 project, as the new version of Godot should be backwards compatible. That being said, we only officially support the latest stable version of Godot and do not have the resources to maintain multiple versions in parallel.

When Godot 4.3 is released, we will update the project to support the new version of Godot.

**Reply by 31** [reply]
How would I make it work with non-backwards-compatible changes?

**Reply by 31** [reply]
Maybe it would help to mention the kind of steps I'm expecting to need to do for incompatible but not *too*-breaking changes... regenerate some glue to that's based on a changed `extension_api.json` or replace a `gdextension_interface.h`, something like that.

**Reply by Splizard** [reply]
As indicated above, you aren't expected to make any changes to use 4.3, it should just work. If you don't want to upgrade when `gd` upgrades, then you can freeze the commit of the project and keep using it and/or fork it as necessary. There isn't currently a simple way to change between Godot versions.

**Reply by 31** [reply]
I'm not all that interested in backward compatibility of Godot to old GDExtensions (I think I was wrong to talk about this as "compatibility", apologies)--there are new classes in 4.3 I'm interested in.

I noticed that `classdb.go` is actually generated by `internal/tool/generate`. Is there a lot of manual cleanup to do outside of running that tool, or something else that makes it not simple?

**Reply by Splizard** [reply]
You can certainly try `go generate` inside of the repo replace the `extension_api.json` with the 4.3 version, `go generate` again and see if it builds.

**Reply by 31** [reply]
Thanks! The initial results are promising, and the compile errors for `PackedVector4Array` are pointing me in the right direction to figure out what hand-written glue is needed on top of that. (Submitted a quick PR to clean up initial reproducibility a bit: https://github.com/grow-graphics/gd/pull/42.)

**Comment by Splizard** [comment, marked as answer]
`grow.graphics/gd` has been updated to target Godot 4.3 so all you need to do now, is use the latest master branch version.

**Reply by Splizard** [reply]
If you need to use an earlier version, you can lock your go.mod to a specific git commit.
